### ISSUE TITLE
Reference @import? (variables and mixins only)

### ISSUE THREAD

**Issue by thybzi** [stylus/stylus#2139]
**TL;DR**: yes, what I want is something similar to [LESS reference import](http://lesscss.org/features/#import-options) to be implemented in Stylus

**A long and boring story:**

Let's say we have the following file structure:
```
|
|-- main
|    |-- button.styl
|    |-- authform.styl
|    `-- mypage.styl
`-- main.styl
```

_main/button.styl_
```styl
// variables
$button-height = 33px
$button-rounding = 4px

// mixins
button__color($base-color)
  background-color: $base-color
  &:hover
    background-color: darken($base-color, 20%)

// styles
.button
  height: $button-height
  border-radius: $button-rounding
  color: white

  .normal
    button__color(#90091e)

  .alternate
    button__color(#bada55)
```

_main/authform.styl_
```styl
// variables
$authform-pad = 5px
$authform-height = $button-height + (2 * $authform-pad)

// styles
.authform
  box-sizing: border-box
  padding: $authform-pad
  height: $authform-height
```

_main/mypage.styl_
```styl
// styles
.mypage

  header
    height: $authform-height

  .someblock .button
    button__color(#906090)
```

_main.styl_
```styl
@import 'main/authform'
@import 'main/button'
@import 'main/mypage'
```

*initial state*
Trying to compile `main.styl` gives obviously bad results such as ```height: $button-height10px;``` for authform etc.

*@import*
Ok, so variables need to be already defined when used.
So, we add ```@import 'button'``` in the beginning of `authform.styl`, and both ```@import 'authform'``` and ```@import 'button'``` to `mypage.styl`
Now, we have all styles compiled correctly, but all authform styles are repeated twice, and button styles are repeated three times.

*@require*
As we may know, @require constructs prevent repeated imports, so we change all @import to @require. Yay, styles are no more repeated!

But... In output css, button styles come first, then authform styles, and then mypage styles.
But the order in `main.css` was: authform, button, mypage.
This order is broken because authform first imports button (and thus outputs its styles), and only after that outputs its own styles.

In this synthetic situation the order seems to be no object. But in real life, the styles order can be just _crucial_, whereas we cannot control it.

*build with no button*
Now let's create one more build which represents some page with authform but _without_ button (let's say the form has only input, but this form must have the same height)

_main/otherpage.styl_
```styl
@require 'authform'

.otherpage

  .myblock
    min-height: $authform-height
```
_nobutton.styl_
```styl
@require 'main/authform'
@require 'main/otherpage'
```

Compiling `nobutton.styl`, we get button styles which we didn't really need in this css build.
The only chance is to split every component file to at least two parts ("styles" file and "vars+mixins" file), and import them separately.

*The solution that should be great*
LESS has a very good approach for aforesaid problems solution: [reference import option](http://lesscss.org/features/#import-options) 

Let us have something similar in Stylus? Something like `@reference`, or `@use`, or `@refer`, or whatever, which would work as "import only variables and mixins from the file, but don't output any styles".

Thank you in advance!

**Comment by kizu** [comment]
Thanks for the issue, we have some plans on enhancing the `@import`/`@require` in Stylus, very close to what Less have, but with a slightly different syntax. Stay tuned! :)

**Comment by thybzi** [comment]
Cool, waiting for updates! Crucial feature for me.

**Comment by gocoffeecup** [comment]
I think I'm facing the same problem after upgrading, either I was just lucky that it accidentally worked before or something changed.

**Comment by ppscvalentin** [comment]
Variables and mixins don't produce output anyway. The cool thing about LESS' `import (reference)` is that it imports everything, meaning you can extend any rules defined, even if the file is a CSS. This is great if you just want parts of whatever someone else wrote. I really like where Stylus is going and I hope this feature makes it in some near version so I can drop LESS and switch to Stylus.

**Comment by FezVrasta** [comment]
Any updates on it? I think something like the `import` feature of ES6 would be awesome.

```
@import { myMixin } from '../mixins'
```

**Comment by TyIsI** [comment]
Any updates on this?

It would be great to be able to write modules that extend base libraries without including the full common base style in every render.

One work-around is to use multi-stage rendering.

[Updated link to LESS documentation](https://lesscss.org/features/#import-atrules-feature-import-options)
